src/mlagility/api/execute_trt.py

```diff
@@ -145,10 +215,51 @@ def format_field(line: str):
     )
     args = parser.parse_args()
 
+    power_readings = []
+    average_power = 0
+
+    # Start power measurement in a separate thread
+    start_event = threading.Event()
+    stop_event = threading.Event()
+    power_thread = threading.Thread(
+        target=measure_power, args=(start_event, stop_event, power_readings, 0.01)
+    )
+    power_thread.start()
```

Comment:
Is it an issue for power measurement that Python GIL only allows one LoC to run at a time? That should mean there is no concurrency between measuring power vs. actually invoking the GPU. It makes me think you may need to run the power monitor in a subprocess to get true concurrency. Otherwise this implementation may under-estimate the power by only measuring power between invocations?

Reply:
Python GIL usually affects 2 compute bound threads trying to run concurrently. In our case the power measuring thread is essentially an I/O bound thread that sleeps for the most part. And I am also able to verify that I am getting all the right samples from the power read as expected.